tuxemon/event/actions/set_code.py

```diff
@@ -0,0 +1,65 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import final
+
+from tuxemon.event.eventaction import EventAction
+from tuxemon.locale import T
+from tuxemon.menu.input import InputMenu
+from tuxemon.tools import open_dialog
+
+
+@final
+@dataclass
+class SetCodeAction(EventAction):
+    """
+    Set a code and checks if it's correct or not.
+    Case Sensible:
+    ATTENTION and AtTenTION are two different words.
+
+    Script usage:
+        .. code-block::
+
+            set_code <code>,<characters>,<variable>
+
+    Script parameters:
+        code: The actual code the player needs to know.
+        characters: The number of characters/digits.
```

Comment:
We could calculate this automatically by counting the number of characters in code. Or did you want to allow values longer than the actual code?

Reply:
Automatically by counting the number of characters is perfect, so we can remove a parameter, that's perfect.